cdk/v2/destinations/webhook/procWorkflow.yaml

```diff
@@ -0,0 +1,82 @@
+bindings:
+  - path: ../../../../v0/util
+  - path: ../../../../v0/destinations/webhook/utils
+  - path: ../../../../constants
+steps:
+  - name: validateInput
+    template: |
+      (
+        $assert(destination.Config.webhookUrl or false, "Invalid URL in destination config");
+        $assert($exists(message.type), "message Type is not present. Aborting message.");
+      )
+  - name: buildHeaderBlock
+    template: |
+      (
+        $defaultHeaders := destination.Config.webhookMethod in ['POST', 'PUT', 'PATCH'] or $not($exists(destination.Config.webhookMethod)) ? {"content-type": "application/json"};
+        $finalHeaders := $merge([$defaultHeaders, $getHashFromArray(destination.Config.headers)]);
+        $messageHeader := $type(message.header) = "object" ? message.header : {};
+        $finalHeaders := $merge([$finalHeaders, $sift($messageHeader, function($v){(
+          $type($v) = "string"
+          )})]);
+        $setContext("headers", $finalHeaders);
+      )
+  - name: deduceEndPoint
+    template: |
+      (
+        $integrationsObjects := $getIntegrationsObj(message, "webhook");
+        $defaultEndpoint := destination.Config.webhookUrl;
+        $finaleEndpoint := [ message.fullPath , $integrationsObjects.fullPath, $defaultEndpoint ][0];
+        $messageAppendPath := [ message.appendPath, $integrationsObjects.appendPath ][0];
+        $finaleEndpoint := $finaleEndpoint & $messageAppendPath;
+        $setContext("endpoint", $finaleEndpoint)
+      )
+  - name: deduceMethod
```

Comment:
move this before buildHeaderBlock and then you can change:
 ```
        $defaultHeaders := destination.Config.webhookMethod in ['POST', 'PUT', 'PATCH'] or $not($exists(destination.Config.webhookMethod)) ? {"content-type": "application/json"};
```
to
```
$defaultHeaders := $context.method in ['POST', 'PUT', 'PATCH'] ?  {"content-type": "application/json"};
```

Reply:
addressed